Sources/KavaPlugin+AnalyticsPluginProtocol.swift

```diff
@@ -0,0 +1,218 @@
+                })
+            
+            case let e where e.self == PlayerEvent.trackChanged:
+                self.messageBus?.addObserver(self, events: [e.self], block: { [weak self] (event) in
+                    guard let strongSelf = self else { return }
+                    PKLog.debug("trackChanged event: \(event)")
+                    if let track = event.selectedTrack {
+                        if (track.title.contains("event.selectedTrack?.title")) {
+                            strongSelf.kavaData.currentCaptionLanguage = event.selectedTrack?.language
+                            strongSelf.sendAnalyticsEvent(event: KavaEventType.captions)
+                        }
+                    }
+                })
+                
+            case let e where e.self == PlayerEvent.error:
+                self.messageBus?.addObserver(self, events: [e.self], block: { [weak self] (event) in
+                    guard let strongSelf = self else { return }
+                    PKLog.debug("PlayerEvent error event: \(event)")
+                    strongSelf.kavaData.errorCode = (event.error?.code)!
+                    strongSelf.sendAnalyticsEvent(event: KavaEventType.error)
+                })
+            default: assertionFailure("all events must be handled")
+            }
+        }
+        
+        self.messageBus?.addObserver(self, events: [AdEvent.error], block: { [weak self] (event) in
+            guard let strongSelf = self else { return }
+            PKLog.debug("AdEvent error event: \(event)")
+            strongSelf.kavaData.errorCode = (event.error?.code)!
+            strongSelf.sendAnalyticsEvent(event: KavaEventType.error)
+        })
+    }
+    
+    /************************************************************/
+    // MARK: Private Implementation
+    /************************************************************/
+    
+    private func handleStateChanged(event: PKEvent) {
+        switch event.newState {
+        case .idle:
+            PKLog.debug("state changed to idle")
+        case .ended:
+            PKLog.info("media ended")
+        case .ready:
+             PKLog.info("media ready")
+            if let _ = bufferingStartTime {
+                self.kavaData.totalBufferingInCurrentInterval += -bufferingStartTime!.timeIntervalSinceNow
+                bufferingStartTime = nil
+            }
+            
+            self.sendMediaLoaded()
+            self.registerToBoundaries()
+        case .buffering:
+            bufferingStartTime = Date()
+        case .error: break
+        case .unknown: break
+        }
+    }
+    
+    private func sendMediaLoaded() {
+        if !self.kavaData.isMediaLoaded {
+            self.kavaData.isMediaLoaded = true
+            
+            sendAnalyticsEvent(event: KavaEventType.impression)
+        }
+    }
+    
+    private func updateDeliveryType(mediaFormat: PKMediaSource.MediaFormat) {
```

Comment:
seems like this should be an API on player, to get the playback stream type

Reply:
we already have it on every PKMediaSource we have MediaFormat just translated it to the convention KAVA plugin needs